src/test/java/io/confluent/connect/jdbc/source/integration/BaseConnectorIT.java

```diff
@@ -0,0 +1,220 @@
+    TestUtils.waitForCondition(
+        () -> assertConnectorAndTasksRunning(name, numTasks).orElse(false),
+        CONNECTOR_STARTUP_DURATION_MS,
+        "Connector tasks did not start in time."
+    );
+    return System.currentTimeMillis();
+  }
+
+  /**
+   * Confirm that a connector with an exact number of tasks is running.
+   *
+   * @param connectorName the connector
+   * @param numTasks      the expected number of tasks
+   * @return true if the connector and tasks are in RUNNING state; false otherwise
+   */
+  protected Optional<Boolean> assertConnectorAndTasksRunning(String connectorName, int numTasks) {
+    try {
+      ConnectorStateInfo info = connect.connectorStatus(connectorName);
+      boolean result = info != null
+          && info.tasks().size() == numTasks
+          && info.connector().state().equals(AbstractStatus.State.RUNNING.toString())
+          && info.tasks().stream().allMatch(s -> s.state().equals(AbstractStatus.State.RUNNING.toString()));
+      return Optional.of(result);
+    } catch (Exception e) {
+      log.warn("Could not check connector state info. Connector might not have started.");
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Create a map of Common connector properties.
+   *
+   * @return : Map of props.
+   */
+  public Map<String, String> getCommonProps() {
+
+    Map<String, String> props = new HashMap<>();
+    props.put("connector.class", JdbcSourceConnector.class.getName());
+    props.put("tasks.max", MAX_TASKS);
+    // license properties
+    props.put("confluent.topic.replication.factor", "1");
+    props.put("confluent.topic.bootstrap.servers", connect.kafka().bootstrapServers());
+
+    // connector-specific properties
+    return props;
+  }
+
+  protected Connection getConnection() throws SQLException {
+    //Class.forName("com.mysql.jdbc.Driver");
+    return DriverManager.getConnection("jdbc:mysql://localhost:3307/db" , "user", "password");
+  }
+
+  protected void assertRecordOffset(int start, int end, ConsumerRecords<byte[], byte[]> records) {
+    for (ConsumerRecord<byte[], byte[]> record : records) {
+      Assert.assertEquals(start++, record.offset());
+    }
+    if(start!=end){
+      throw new AssertionError("Invalid number of records found");
+    }
+  }
+
+  protected void addConnectionProperties(Map<String, String> props) {
+    props.put("connection.password", "password");
+    props.put("connection.user", "user");
+    props.put("connection.url", "jdbc:mysql://localhost:3307/db");
+    props.put("dialect.name", "MySqlDatabaseDialect");
+    props.put("value.converter", JsonConverter.class.getName());
+    props.put("topic.prefix",  "sql-");
+  }
+
+  protected void dropTableIfExists(String kafkaTopic, Connection connection) throws SQLException {
+    Statement st = connection.createStatement();
+    st.executeQuery("drop table if exists " + kafkaTopic);
+  }
+
+  protected void createTable(Connection connection) throws SQLException {
```

Comment:
```suggestion
  protected void createTable(String tableName) throws SQLException {
```
Connection object can be a member variable in this class and access in the IT tests.
Also, change `getConnection()` -> `initializeJdbcConnection()`.
Just a suggestion, we can discuss...